Makefile

```diff
@@ -523,10 +524,12 @@ kind-image: ## Build cilium-dev docker image and import it into kind.
 	@echo "  DEPLOY image to kind ($(LOCAL_AGENT_IMAGE))"
 	$(QUIET)$(CONTAINER_ENGINE) push $(LOCAL_AGENT_IMAGE)
 	$(QUIET)kind load docker-image $(LOCAL_AGENT_IMAGE)
+ifeq ($(SKIP_OPERATOR),"false")
```

Comment:
IMO - we should split the building of the kind images and their deploying. 

This means we'd be more declarative with make targets vs adding yet another special case flag. 

For example if we split this into:

`make kind-images`
`make kind-image-deploy-agent`
`make kind-image-deploy-operator`

And then if we want to tie them all together, just call them all in series with a `make kind-build-and-deploy` or something similar. 

This is touching on the back and forth we are dwindling down on in #21108

Reply:
How far should we go on this? In particular, I'm trying to avoid compiling or loading the operator at all.

`make kind-image-agent`
`make kind-image-operator`
`make kind-images`
`make kind-image-deploy-agent`
`make kind-image-deploy-operator`
`make kind-image-deploy`

I guess make targets are cheap, the list above seems reasonable enough.

Incidentally, the thing that I like about the current proposal is that I rarely do operator development, so I can just put `SKIP_OPERATOR` in my .bashrc and forget about it. That said, adding lots of `SKIP_FOO` all over the place will get out-of-hand over time. I'll play around with make targets a bit and respin.

Reply:
>In particular, I'm trying to avoid compiling or loading the operator at all.

>Incidentally, the thing that I like about the current proposal is that I rarely do operator development, so I can just put SKIP_OPERATOR in my .bashrc and forget about it.

+1 This mirrors my dev workflow as well. make targets are kind of hard to remember. Deepesh had added a help option to list all the targets, but I'm not sure if that list is being updated.

Reply:
The help target is tied directly to the make targets. As long as the new targets have a comment next to them in the same style as the existing targets, then these will all show up in `make help`.